app/uk/gov/hmrc/partnershipidentificationfrontend/service/ValidationOrchestrationService.scala

```diff
@@ -27,56 +27,50 @@ class ValidationOrchestrationService @Inject()(partnershipIdentificationService:
                                                validatePartnershipInformationService: ValidatePartnershipInformationService
                                               )(implicit ec: ExecutionContext) {
 
-  def orchestrate(journeyId: String, businessVerificationCheck: Boolean)(implicit hc: HeaderCarrier): Future[ValidationResponse] =
+  def orchestrate(journeyId: String, businessVerificationCheck: Boolean)(implicit hc: HeaderCarrier): Future[ValidationResponse] = {
     partnershipIdentificationService.retrievePartnershipInformation(journeyId).flatMap {
-      case Some(PartnershipInformation(Some(SaInformation(sautr, postcode)), Some(_))) =>
-        validatePartnershipInformationService.validateIdentifiers(sautr, postcode).flatMap {
-          _ =>
-            for {
-              _ <- partnershipIdentificationService.storeIdentifiersMatch(journeyId, identifiersMatch = false)
-              _ <- if (businessVerificationCheck) {
-                partnershipIdentificationService.storeBusinessVerificationStatus(journeyId, BusinessVerificationUnchallenged)
-              } else {
-                Future.successful(())
-              }
-              _ <- partnershipIdentificationService.storeRegistrationStatus(journeyId, RegistrationNotCalled)
-            } yield
-              IdentifiersMismatch
-        }
-      case Some(PartnershipInformation(Some(SaInformation(sautr, postcode)), _)) =>
-        validatePartnershipInformationService.validateIdentifiers(sautr, postcode).flatMap {
-          identifiersMatch =>
-            if (identifiersMatch) {
-              partnershipIdentificationService.storeIdentifiersMatch(journeyId, identifiersMatch = true).map {
-                _ => IdentifiersMatched
-              }
-            }
-            else {
-              for {
-                _ <- partnershipIdentificationService.storeIdentifiersMatch(journeyId, identifiersMatch = false)
-                _ <- if (businessVerificationCheck) {
-                  partnershipIdentificationService.storeBusinessVerificationStatus(journeyId, BusinessVerificationUnchallenged)
-                } else {
-                  Future.successful(())
+      case Some(PartnershipInformation(optSaInformation, optCompanyProfile)) => for {
+        identifiersMatch <- optSaInformation match {
+          case Some(SaInformation(sautr, postcode)) =>
+            validatePartnershipInformationService.validateIdentifiers(sautr, postcode) flatMap {
+              identifiersMatch =>
```

Comment:
The identifier "identifiersMatch" is used in a number of circumstances which is a little confusing. Perhaps, the output of the SA matching, defined in line 36 could be changed to "saIdentifiersMatch". I know it is a bit tricky as the meaning of matching varies between the two general types of partnership.